### PR TITLE
[Flake8] Fix `Errno::ENOENT` on multiple configurations

### PR DESCRIPTION
**Pull request by ybiquitous**
> Explain a summary, purpose, or background for this change.

When there are multiple configurations (`setup.cfg` and `tox.ini`) in a repository, the Flake8 runner raises `Errno::ENOENT`.

```
/home/analyzer_runner/runners/lib/runners/processor/flake8.rb:70:in `delete': No such file or directory @ apply2files - /home/analyzer_runner/.config/flake8 (Errno::ENOENT)
    from /home/analyzer_runner/runners/lib/runners/processor/flake8.rb:70:in `block in prepare_config'
    from /home/analyzer_runner/runners/lib/runners/processor/flake8.rb:68:in `each'
    from /home/analyzer_runner/runners/lib/runners/processor/flake8.rb:68:in `prepare_config'
    from /home/analyzer_runner/runners/lib/runners/processor/flake8.rb:38:in `setup'
    from /home/analyzer_runner/runners/lib/runners/harness.rb:58:in `block (3 levels) in run'
    from /home/analyzer_runner/runners/lib/runners/shell.rb:58:in `block in chdir'
    from /home/analyzer_runner/runners/lib/runners/shell.rb:56:in `chdir'
    from /home/analyzer_runner/runners/lib/runners/shell.rb:56:in `chdir'
    from gems/forwardable-1.3.2/lib/forwardable.rb:238:in `chdir'
    from /home/analyzer_runner/runners/lib/runners/processor.rb:147:in `in_root_dir'
    from /home/analyzer_runner/runners/lib/runners/harness.rb:51:in `block (2 levels) in run'
    from /home/analyzer_runner/runners/lib/runners/workspace.rb:34:in `block in open'
    from /home/analyzer_runner/runners/lib/runners/workspace.rb:81:in `prepare_ssh'
    from /home/analyzer_runner/runners/lib/runners/workspace.rb:19:in `open'
    from /home/analyzer_runner/runners/lib/runners/harness.rb:34:in `block in run'
    from /home/analyzer_runner/runners/lib/runners/harness.rb:97:in `ensure_result'
    from /home/analyzer_runner/runners/lib/runners/harness.rb:32:in `run'
    from /home/analyzer_runner/runners/lib/runners/cli.rb:69:in `run'
    from /home/analyzer_runner/runners/exe/runners:25:in `<main>'
```

> Refer related issues or pull requests (e.g. `Close #123` or `None`).

None.

> Check the following if needed.

- [x] Add a new entry to [CHANGELOG.md](https://github.com/sider/runners/blob/master/CHANGELOG.md) if this change is notable.

### CODE REVIEW

File: lib/runners/processor/flake8.rb [hunk 1]
```diff
@@ -66,14 +66,11 @@ def analyze(changes)
 
     def prepare_config
       # @see https://flake8.pycqa.org/en/latest/user/configuration.html
-      if (current_dir / '.flake8').exist?
+      case
+      when (current_dir / '.flake8').exist?
+        File.delete DEFAULT_CONFIG_PATH
+      when current_dir.glob('{setup.cfg,tox.ini}').any? { |f| f.read.match?(/^\[flake8\]$/m) }
```
Comment (ybiquitous) [inline lib/runners/processor/flake8.rb:72]: > __[RuboCop]__ Duplicate branch body detected. ([view](https://sider.review/gh/repos/202486306/pulls/2281#issue-5516933))
>
> <details>
>
> | Rule | Severity |
> | ---- | -------- |
> | `Lint/DuplicateBranch` | `warning` |
>
> References:
> - <https://docs.rubocop.org/rubocop/cops_lint.html#lintduplicatebranch>
>
> You can [close this issue](https://sider.review/gh/repos/202486306/pulls/2281#issue-5516933) if no need to fix it. [Learn more](https://help.sider.review/getting-started/working-with-issues).
>
> </details>

I think it acceptable.

**Review by yykamei — APPROVED** [review]
:shipit:

**Comment by ybiquitous** [comment]
I'll fix the RuboCop test failure via another PR.

**Comment by ybiquitous** [comment]
> I'll fix the RuboCop test failure via another PR.

Done #2282

**Pull request merged**